### PR TITLE
Feature/issue 119

### PR DESCRIPTION
**Pull request by nlenssen2013**
Github Issue: #119 

### Description

Currently Variables in GPM point to 'phony_dims' which inhibits subsetting_with_bbox. Variables in GPM file need the dimensions in the attribute DimensionName

### Overview of work done

Each variable is checked for and any unique DimensionNames will be created into a new dimension and the variable will point to that new dimension instead of the 'phony_dim' that is there currently.

### Overview of verification done

Test that the new variables don't have any dimensions with 'phony' in them.

### Overview of integration done

_Explain how this change was integration tested. Provide screenshots or logs if appropriate. An example of this would be a local Harmony deployment._

## PR checklist:

* [x ] Linted
* [ x] Updated unit tests
* [ x] Updated changelog
* [ ] Integration testing

_See [Pull Request Review Checklist](../CONTRIBUTING.md#reviewing) for pointers on reviewing this pull request_

### CODE REVIEW

**Review by jamesfwood — APPROVED** [review]

File: podaac/subsetter/subset.py [hunk 1]
```diff
@@ -532,10 +533,9 @@ def compute_time_variable_name(dataset: xr.Dataset, lat_var: xr.Variable) -> str
         var_name_time = var_name.strip(GROUP_DELIM).split(GROUP_DELIM)[-1]
         if len(dataset[var_name].squeeze().dims) == 0:
             continue
-        if 'time' == var_name_time.lower() and dataset[var_name].squeeze().dims[0] in lat_var.squeeze().dims:
+        if ('time' == var_name_time.lower() or 'timeMidScan' == var_name_time) and dataset[var_name].squeeze().dims[0] in lat_var.squeeze().dims:
             return var_name
 
-    # then check if any variables have 'time' in the string if the above loop doesn't return anything
```
Comment (danielfromearth) [inline podaac/subsetter/subset.py:538]: Was this comment intentionally removed?  Seems like keeping it may still be helpful?

File: podaac/subsetter/subset.py [hunk 2]
```diff
@@ -1179,6 +1182,9 @@ def subset(file_to_subset: str, bbox: np.ndarray, output_file: str,
         lon_var_names = [var.replace('/', GROUP_DELIM) for var in lon_var_names]
         time_var_names = [var.replace('/', GROUP_DELIM) for var in time_var_names]
 
+        if file_extension == 'HDF5':
```
Comment (danielfromearth) [inline podaac/subsetter/subset.py:1185]: basically the same comment as above, on line 1039

File: podaac/subsetter/subset.py [hunk 3]
```diff
@@ -1204,8 +1210,9 @@ def subset(file_to_subset: str, bbox: np.ndarray, output_file: str,
             time_var_names=time_var_names
         )
         start_date = None
-        if file_extension == 'he5' and (min_time or max_time):
-            dataset, start_date = convert_to_datetime(dataset, time_var_names)
+        if file_extension in ['he5', 'HDF5'] and (min_time or max_time):
```
Comment (danielfromearth) [inline podaac/subsetter/subset.py:1213]: basically the same comment again as above, on line 1039

File: tests/test_subset.py [hunk 4]
```diff
@@ -1811,7 +1812,7 @@ def test_temporal_he5file_subset(data_dir, subset_output_dir):
             if 'BRO' in i:
                 assert any('utc' in x.lower() for x in time_var_names)
 
-            dataset, _ = subset.convert_to_datetime(dataset, time_var_names)
+            dataset, _ = subset.convert_to_datetime(dataset, time_var_names, i[1].split('.')[-1])
```
Comment (danielfromearth) [inline tests/test_subset.py:1815]: For getting extensions, it seems to me like we should use standard library methods to avoid edge cases (e.g., see https://stackoverflow.com/a/541394).  Using the `os` library would probably be the simplest and most straightforward here. (although `pathlib` seems to me to be the most robust, I think it would require a larger refactoring)

File: podaac/subsetter/gpm_cleanup.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
+"""
```
Comment (danielfromearth) [inline podaac/subsetter/gpm_cleanup.py:1]: Question for all: should we have a separate folder or module that has all collection- (or project-) specific logic?  I can't immediately think of a way to generalize this, and if we need some one-off logic, perhaps it should be separated a bit more from the generic subsetter module files?

File: podaac/subsetter/subset.py [hunk 6]
```diff
@@ -1036,7 +1036,10 @@ def convert_to_datetime(dataset: xr.Dataset, time_vars: list) -> Tuple[xr.Datase
     """
 
     for var in time_vars:
-        start_date = datetime.datetime.strptime("1993-01-01T00:00:00.00", "%Y-%m-%dT%H:%M:%S.%f")
+        if file_extension == 'HDF5':
```
[the remark this answers is not in the document]
Comment (nlenssen2013) [reply podaac/subsetter/subset.py:1039]: Thanks for going through this @danielfromearth. The solution that I'm targeting now is to have a time_converting.py module. Have converting functions for tai93 and GPS time. As of now, we know that xarray doesn't decode times for he5 and HDF5 file extensions. So instead of checking extension having a dictionary of the collections we know need what time conversion? Thoughts?

[the remark this answers is not in the document]
Comment (danielfromearth) [reply podaac/subsetter/subset.py:1039]: That sounds like it would work. However, thinking about this more makes me worried about hard-coding a dictionary of collection names into the code.  Would it be reasonable to check the first (or first few) time value, and use that to determine whether (and which) conversion is needed?

[the remark this answers is not in the document]
Comment (nlenssen2013) [reply podaac/subsetter/subset.py:1041]: @danielfromearth Yeah, and use the min/max times inputed to get which start date. Will put that into the next PR if that's alright with you.

File: podaac/subsetter/time_converting.py
Comment (nlenssen2013) [inline podaac/subsetter/time_converting.py:1]: Added this module since some of the files coming in will need other functions for time converting

File: podaac/subsetter/group_handling.py [hunk 7]
```diff
@@ -242,6 +242,13 @@ def walk_h5py(data_new, group):
 
     walk_h5py(data_new, data_new.name)
```
Comment (nlenssen2013) [inline podaac/subsetter/group_handling.py:244]: Getting the instrument via attributes

**Review by jamesfwood — APPROVED** [review]
